File: Samples/JotFormTest/Classes/Controller/CreateFormsViewController.m
//
//  CreateFormsViewController.m
//  JotFormTest
//
//  Created by Curtis Stilwell on 10/20/14.
//  Copyright (c) 2014 wang. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "CreateFormsViewController.h"
#import "Common.h"
#import "SharedData.h"
#import "SVProgressHUD.h"

@implementation CreateFormsViewController

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil {
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
}

#pragma mark - User definition method

- (void)createForms {
    NSError *error = nil;
    NSData *objectData = [@"{\"questions\":[{\"type\":\"control_head\"}]}" dataUsingEncoding:NSUTF8StringEncoding];
    NSDictionary *json = [NSJSONSerialization JSONObjectWithData:objectData
                                                         options:NSJSONReadingMutableContainers
                                                           error:&error];
  
    [[SharedData sharedData].apiClient createForms:json onSuccess:^(id result){
   
    } onFailure:^(NSError *error) {

    }];
}

#pragma mark - IBAction

- (IBAction)createFormsViewClicked:(id)sender {
    [self createForms];
}


@end

File: Samples/JotFormTest/Classes/Controller/CreateFormsViewController.swift
//
//  CreateFormsViewController.swift
//  JotFormTest
//
//  Created by Curtis Stilwell on 2/4/18.
//  Copyright © 2018 wang. All rights reserved.
//

import Foundation

class CreateFormsViewController: UIViewController {
    override func viewDidLoad() {
        // Do any additional setup after loading the view from its nib.
        title = "Create Forms"
        
    }
    
    func createForms() {
        let objectData: Data? = "{\"questions\":[{\"type\":\"control_head\"}]}".data(using: .utf8)
        if let json = (try? JSONSerialization.jsonObject(with: objectData ?? Data(), options: .mutableContainers)) as? [String: Any] {
            
            SharedData.sharedData.apiClient?.createForms(json, onSuccess: {(_ result: Any) -> Void in
            }, onFailure: {(_ error: Error?) -> Void in
            })
        }
    }

    @IBAction func createFormsViewClicked(_ sender: Any) {
        createForms()
    }
}
